framework/src/utils/RankThreeTensor.C

```diff
@@ -185,23 +165,18 @@ RankThreeTensor
 RankThreeTensor::operator+(const RankThreeTensor & b) const
 {
   RankThreeTensor result;
-  const RankThreeTensor & a = *this;
 
-  for (unsigned int i = 0; i < N; ++i)
-    for (unsigned int j = 0; j < N; ++j)
-      for (unsigned int k = 0; k < N; ++k)
-        result(i, j, k) = a(i, j, k) + b(i, j, k);
+  for (unsigned int i = 0; i < N3; ++i)
+    result._vals[i] = _vals[i] + b._vals[i];
```

Comment:
On linux g++ we get no vectorization - mostly due to (potential) pointer aliasing that the compiler can't prove doesn't occur.  Not sure on mac.